bundles/core/src/main/java/com/adobe/cq/commerce/core/components/internal/models/v1/page/PageImpl.java

```diff
@@ -0,0 +1,156 @@
+/*******************************************************************************
+ *
+ *    Copyright 2019 Adobe. All rights reserved.
+ *    This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License. You may obtain a copy
+ *    of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software distributed under
+ *    the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ *    OF ANY KIND, either express or implied. See the License for the specific language
+ *    governing permissions and limitations under the License.
+ *
+ ******************************************************************************/
+
+package com.adobe.cq.commerce.core.components.internal.models.v1.page;
+
+import java.util.Calendar;
+import java.util.Map;
+
+import javax.annotation.PostConstruct;
+import javax.inject.Inject;
+
+import org.apache.sling.api.SlingHttpServletRequest;
+import org.apache.sling.models.annotations.Model;
+import org.apache.sling.models.annotations.Via;
+import org.apache.sling.models.annotations.injectorspecific.Self;
+import org.apache.sling.models.annotations.via.ResourceSuperType;
+
+import com.adobe.cq.commerce.core.components.models.page.Page;
+import com.adobe.cq.export.json.ComponentExporter;
+import com.adobe.cq.wcm.core.components.models.NavigationItem;
+import com.day.cq.commons.inherit.HierarchyNodeInheritanceValueMap;
+import com.day.cq.commons.inherit.InheritanceValueMap;
+
+@Model(
+    adaptables = SlingHttpServletRequest.class,
+    adapters = { Page.class },
+    resourceType = PageImpl.RESOURCE_TYPE)
+public class PageImpl implements Page {
+
+    protected static final String RESOURCE_TYPE = "core/cif/components/structure/page/v1/page";
+    private static final String STORE_CODE_PROPERTY = "cq:magentoStore";
```

Comment:
Should we consider renaming this JCR property to cq:magentoStoreView sometime? In Magento there a store can have multiple store views. I think we actually use this property for the store view.

Reply:
This property will most probably disappear with CIF-1086.